src/distributions.jl

```diff
@@ -18,21 +18,34 @@ function default_range(m::Distributions.MixtureModel, alpha = 0.0001)
             maxval = thismax
         end
     end
-    minval, maxval
+    promote(minval, maxval)
 end
 
 yz_args(dist) = default_range(dist)
-yz_args(dist::Distribution{N, T}) where N where T<:Discrete = (UnitRange(Int.(default_range(dist))...),)
+function yz_args(dist::DiscreteUnivariateDistribution)
+    minval, maxval = extrema(dist)
+    if isfinite(minval) && isfinite(maxval)  # bounded
+        sup = support(dist)
+        return sup isa AbstractVector ? (sup,) : ([sup...],)
+    else  # unbounded
+        return (UnitRange(default_range(dist)...),)
+    end
+end
 
 # this "user recipe" adds a default x vector based on the distribution's μ and σ
 @recipe function f(dist::Distribution)
-    if dist isa Distribution{Univariate,Discrete}
-        seriestype --> :scatterpath
+    if dist isa Distribution{Univariate,Discrete} && get(plotattributes, :seriestype, :none) === :none
```

Comment:
Maybe a bit more standard:
```suggestion
    if dist isa DiscreteUnivariateDistribution && get(plotattributes, :seriestype, :none) === :none
```